packages/compass-query-bar/src/components/query-bar/query-bar.tsx

```diff
@@ -49,74 +60,218 @@ const openQueryHistoryStyles = cx(
   focusRingStyles
 );
 
+const rowStyles = css({
+  display: 'flex',
+  flexGrow: 1,
+});
+
 type QueryBarProps = {
+  autoPopulated: boolean;
   buttonLabel?: string;
   expanded: boolean;
-  isQueryOptionsExpanded?: boolean;
-  valid: boolean;
+  filterValid: boolean;
+  layout?: QueryBarLayout;
+  onApply: () => void;
+  onChangeQueryOption: (queryOption: QueryOption, value: string) => void;
+  onReset: () => void;
   queryState: 'apply' | 'reset';
+  refreshEditorAction: () => void;
+  schemaFields: string[];
+  serverVersion: string;
   showQueryHistoryButton?: boolean;
   toggleExpandQueryOptions: () => void;
   toggleQueryHistory: () => void;
+  valid: boolean;
 };
 
 export const QueryBar: React.FunctionComponent<QueryBarProps> = ({
+  autoPopulated,
   buttonLabel = 'Apply',
   expanded: isQueryOptionsExpanded = false,
-  valid: isQueryValid,
+  filterValid: isFilterValid,
+  layout = [
+    'filter',
+    'project',
+    ['sort', 'maxTimeMS'],
+    ['collation', 'skip', 'limit'],
+  ],
+  onApply: _onApply,
+  onChangeQueryOption,
+  onReset: _onReset,
   queryState,
+  refreshEditorAction,
+  schemaFields,
+  serverVersion,
   showQueryHistoryButton = true,
   toggleExpandQueryOptions,
-  toggleQueryHistory,
+  toggleQueryHistory: _toggleQueryHistory,
+  valid: isQueryValid,
+
+  // This form uses extra props from the store for placeholders and
+  // input values for the query options. `filterValue`, `sortValue`, etc.
+  ...props
```

Comment:
In the next pr we'll type out all of the various query option props:
`filterValue`, `filterPlaceholder`, `sortValue`, etc. For this reason there are 3 `any` in this file. I could do it in this pr  if preferred.